pkg/models/repository_configuration.go

```diff
@@ -11,8 +11,8 @@ import (
 type RepositoryConfiguration struct {
 	Base
 	Name           string         `json:"name" gorm:"default:null"`
-	Versions       pq.StringArray `json:"version" gorm:"type:text[],default:null"`
-	Arch           string         `json:"arch" gorm:"default:''"`
+	Versions       pq.StringArray `json:"version" gorm:"type:text[],default:['any']"`
```

Comment:
Not sure if you saw my last comment before you pushed your latest change, but right now it's possible to create/update a repository with `[]` as the distribution_versions. If you can't get this "default" field tag to work, you may need to move filling this default to be part of validation (or as part of the beforecreate/beforeupdate hooks).

Reply:
If you do go that route, I think the `createVersionArray` function in seeds could go back to what you had previously.